### ISSUE TITLE
Expose `__matmul__` for matrix multiplication

### ISSUE THREAD

**Issue by mwtoews** [rasterio/affine#107]
Affine has a `__mul__` magic method for matrix multiplication with the `*` operator.

Python 3.5 added the `@` operator for matrix multiplication (matmul), with several magic methods added to the [object data model](https://docs.python.org/3/reference/datamodel.html). Probably `__matmul__` should be the primary definition for `__mul__`. Don't implement `__imatmul__` (`@=`) since Affine should be immutable. Maybe implement `__rmatmul__` if order is swapped?

**Comment by mwtoews** [comment]
I should mention that I have some work in progress for this one, which will be ready soon.

**Comment by sgillies** [comment]
@mwtoews Yes, we should implement `__rmatmul__` for a tuple on the left-hand side. I mentioned this in #48.

**Comment by sgillies** [comment]
In #48 we've agreed that `__rmatmul__` and `__rmul__` should raise in all cases.

**Comment by mwtoews** [comment]
This hasn't been done yet
```pycon
>>> from affine import identity
>>> identity @ identity
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unsupported operand type(s) for @: 'Affine' and 'Affine'
```
but I have an ambitious PR almost ready...
